custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -115,3 +116,53 @@ def _validate_content_type_header(header):
             response.headers["Content-Type"] = content_type
 
         return response, response_status
+
+    def do_transform(self, logger=None):
+        response_status = HTTP_200_OK
+
+        if self._target_type != TargetType.TRANSFORM.value:
+            wrong_target_type_error_message = "transform only applicable to {} target type. This project has target type {}.".format(
+                TargetType.TRANSFORM.value, self._target_type
+            )
+            if logger is not None:
+                logger.error(wrong_target_type_error_message)
+            response_status = HTTP_422_UNPROCESSABLE_ENTITY
+            return {"message": "ERROR: " + wrong_target_type_error_message}, response_status
+
+        file_key = "X"
+        filestorage = request.files.get(file_key)
+
+        if not filestorage:
+            wrong_key_error_message = (
+                "Samples should be provided as a csv, mtx, or arrow file under `{}` key.".format(
+                    file_key
+                )
+            )
+            if logger is not None:
+                logger.error(wrong_key_error_message)
+            response_status = HTTP_422_UNPROCESSABLE_ENTITY
+            return {"message": "ERROR: " + wrong_key_error_message}, response_status
+        else:
+            if logger is not None:
+                logger.debug("Filename provided under X key: {}".format(filestorage.filename))
+
+        _, file_ext = os.path.splitext(filestorage.filename)
+        with tempfile.NamedTemporaryFile(suffix=file_ext) as f:
+            filestorage.save(f)
+            f.flush()
+            out_data = self._predictor.transform(f.name)
+
+        # float32 is not JSON serializable, so cast to float, which is float64
```

Comment:
It would be better, IMHO, to use the `default_handler` argument of the `to_json` function?

Reply:
I'm reading the description of that arg but still not totally clear on how to use it- got an example handy? thanks!